scripts/devops_tasks/tox_harness.py

```diff
@@ -446,12 +320,19 @@ def prep_and_run_tox(targeted_packages, parsed_args, options_array=[]):
             replace_dev_reqs(dependency_tools_path, package_dir)
             os.environ["TOX_PARALLEL_NO_SPINNER"] = "1"
 
-        inject_custom_reqs(
-            destination_dev_req, parsed_args.injected_packages, package_dir
-        )
+        inject_custom_reqs(destination_dev_req, parsed_args.injected_packages, package_dir)
 
         if parsed_args.tox_env:
-            tox_execution_array.extend(["-e", parsed_args.tox_env])
+            filtered_tox_environment_set = filter_tox_environment_string(parsed_args.tox_env, package_name)
```

Comment:
Compare package name, exclusion list, and incoming tox environments -> generate an amended list with the skipped environments excluded.